apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimTransformerV2.java

```diff
@@ -316,43 +182,41 @@ private static List<Claim.DiagnosisComponent> getDiagnosis(
-                              .setCoding(
-                                  Collections.singletonList(
-                                      new Coding(icdSystem, diagnosisCode.getDiagCd2(), null))));
-
-              if (diagnosisCode.getDiagPoaInd() != null) { // Present on Admission
-                component.setOnAdmission(
-                    new CodeableConcept(
-                        new Coding(
-                            BBCodingSystems.CLM_POA_IND,
-                            diagnosisCode.getDiagPoaInd().toLowerCase(Locale.ROOT),
-                            null)));
-              }
-
-              if (claimGroup.getAdmitDiagCode() != null
-                  && codesAreEqual(claimGroup.getAdmitDiagCode(), diagnosisCode.getDiagCd2())) {
-                component.setType(
-                    List.of(
-                        new CodeableConcept()
-                            .setCoding(
-                                List.of(
-                                    new Coding(
-                                        ExDiagnosistype.ADMITTING.getSystem(),
-                                        ExDiagnosistype.ADMITTING.toCode(),
-                                        ExDiagnosistype.ADMITTING.getDisplay())))));
+              Claim.DiagnosisComponent component;
+
+              if (Strings.isNotBlank(diagnosisCode.getDiagCd2())) {
+                component =
+                    new Claim.DiagnosisComponent()
+                        .setSequence(diagnosisCode.getRdaPosition())
+                        .setDiagnosis(createCodeableConcept(icdSystem, diagnosisCode.getDiagCd2()));
+
+                if (Strings.isNotBlank(diagnosisCode.getDiagPoaInd())) { // Present on Admission
+                  component.setOnAdmission(
+                      createCodeableConcept(
+                          BBCodingSystems.FISS.DIAG_POA_IND,
+                          diagnosisCode.getDiagPoaInd().toLowerCase()));
+                }
+
+                if (codesAreEqual(claimGroup.getAdmitDiagCode(), diagnosisCode.getDiagCd2())) {
+                  component.setType(List.of(createCodeableConcept(ExDiagnosistype.ADMITTING)));
+                }
+              } else {
+                component = null;
               }
 
               return component;
             })
+        .filter(Objects::nonNull)
         .sorted(Comparator.comparing(Claim.DiagnosisComponent::getSequence))
         .collect(Collectors.toList());
   }
 
+  /**
+   * Parses data from the {@link RdaFissClaim} to create a {@link Claim.ProcedureComponent} list.
+   *
+   * @param claimGroup The {@link RdaFissClaim} object to parse data from.
+   * @return The {@link Claim.ProcedureComponent} object list created from the parsed data.
+   */
   private static List<Claim.ProcedureComponent> getProcedure(
       RdaFissClaim claimGroup, boolean isIcd9) {
     final String icdSystem = isIcd9 ? IcdCode.CODING_SYSTEM_ICD_9 : IcdCode.CODING_SYSTEM_ICD_10;
```

Comment:
@KyRobbins The procedure codes need to change per Bill Harty's latest feedback.  We should use `IcdCode.CODING_SYSTEM_ICD_10_MEDICARE` and create a new one just like that in the IcdCode class for `IcdCode.CODING_SYSTEM_ICD_9_MEDICARE`

Reply:
Oh great thought, yes we should update the SAMHSA matcher to filter on this new procedure codes.  I think the BFD might have already updated that logic for EOBs, but we'll have to follow the same path for Claim/ClaimResponse